README.md

```diff
@@ -299,6 +299,39 @@ const MyStyledDiv = glamorous.div(
 <MyStyledDiv /> // styles applied: {padding-top: 1, padding-right: 2, padding-bottom: 3, padding-left: 4} and anything coming from `extra-thing`.
 ```
 
+##### withProps
+
+The `glamorousComponentFactory` has a `withProps` property, which is another
+factory function accepting any number of property object arguments to be applied
+to the `GlamorousComponent`. Just like styles, these can either be property objects
+or functions wich are invoked with `props` on every render and return property objects.
+
+```jsx
+const MyStyledInput = glamorous.input.withProps({type: 'text'})({
+  padding: 10
+})
+
+const MySmartStyledInput = glamorous(MyStyledInput).withProps(
+  props => {
+    if (props.creditCard) {
+      return {
+        type: 'credit-card',
+        pattern: /([0-9]{4}){4}/
+      }
+    }
+    return null
+  }
+)
+
+<MyStyledInput />
+// renders: <Input type="text" />
+// styles applied: {padding: 10}
+
+<MyStyledValidatedInput creditCard />
+// renders: <Input type="credit-card" pattern="/([0-9]{4}){4}/" />
```

Comment:
This should render `<input` rather than `<Input`, also I'm pretty sure that `credit-card` is not a valid `type`. Which is why the pattern is good :)